lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -65,15 +66,15 @@ sub _get_database {
     my ($request) = @_;
     my $creds = $request->{_auth}->get_credentials('setup');
 
-    return [ 401,
+    return [ HTTP_UNAUTHORIZED,
              [ 'WWW-Authenticate' => 'Basic realm="LedgerSMB"',
                'Content-Type' => 'text/text; charset=UTF-8' ],
              [ 'Please enter your credentials' ] ]
         if ! defined $creds->{password};
 
     # Ideally this regex should be configurable per instance, and possibly per admin user
     # for now we simply use a fixed regex. It will cover many if not most use cases.
-    return [ 454,
+    return [ HTTP_454,
```

Comment:
I will defer to greater wisdom for not having any HTTP experience.  The lack of an allowed username or of passwords/keys must have a standard answer.  It looks to me like a authentication/authorization issue.